2.0 Blog post (#61)
btcpayserver/blog: blog/_posts/btcpay-server-2-0.md

---
title: "BTCPay Server 2.0: our biggest update yet!"
date: 2024-10-29
author: Pavlenex
category:
  - "releases"
tags:
  - "btcpay-server"
  - "release"
  - "2.0"

> coverImage: "/images/btcpay2-0-featured.png"

Author: Needs a featured image.

Commitment to stability meant that BTCPay evolved organically, always maintaining backward compatibility. While this benefits users, it has placed increasing pressure on our developers and plugin creators, making it harder and slower to meet new expectations, a challenge known as the "technical debt tax"; that we will now eliminate.

During the transition from 1.0 to 2.0, larger instances may experience **a few minutes of database migration**. Once this initial migration is complete, BTCPay will be fully operational, though it will continue migrating some data in the background. For instance, some invoices may not immediately appear in reports or searches, but newer data is migrated first, so we expect minimal disruption. 

As an example, our mainnet demo database, containing around 400,000 invoices, took one to two minutes to start and about 15-20 minutes to fully update all invoices in the background.

## 🔄 Updating to 2.0

// TODO 

> 

Author: We need to update this section upon we sync cc @NicolasDorier @rockstardev

Reviewer: Sure thing, I'll be back in action tomorrow morning; here is proposed section text we can discuss:

```suggestion
Updating to version BTCPay Server 2.0 is a one-way process with no option for rolling back. Because of that, we're making it opt-in, requiring that you [SSH into your server](https://docs.btcpayserver.org/FAQ/ServerSettings/#how-to-ssh-into-my-btcpay-running-on-vps) and run two simple commands:

- git fetch && git checkout 2.0
- btcpay-update.sh

Please note that the database migration might take a few minutes, depending on the size of your installation. Enjoy BTCPay 2.0 and let us know your thoughts!
```

- If you’re a **plugin developer**, make sure your plugins are 2.0 compatible. To start, bump Bump plugin to 2.0 submodule and fix errors
- If you’re using **Translations**, report any missing strings [here](https://github.com/btcpayserver/btcpayserver/discussions/6295)

## 🌍 Interface localization

BTCPay 2.0 introduces a **full interface translation**. While customer-facing parts of the UI have been multilingual for some time, this update allows you to translate the entire BTCPay Server back office with ease. All translation strings are in a single text box, easy to copy over to AI tools like ChatGPT, making the process of translating strings and pasting them back super-easy. This gives you full control to localize your BTCPay Server in any language quickly and efficiently.

If you come across any missing strings, please report them [here](https://github.com/btcpayserver/btcpayserver/discussions/6295). We’ll continue to refine and complete all translations in the next few minor releases, and we’re adding several predefined languages to further enhance your experience.

> ![](/images/2-0-Translation.png)

Author: Needs an image.

Author: waiting for @TChukwuleta

---
title: "BTCPay Server 2.0: our biggest update yet!"
date: 2024-10-29
author: Pavlenex
category:
  - "releases"
tags:
  - "btcpay-server"
  - "release"
  - "2.0"
coverImage: "/images/btcpay2-0-featured.png"
---

We are thrilled to announce the release of **BTCPay Server 2.0**, our most significant update since the project began in 2017! 

This version is packed with new features, bug fixes, and crucial backend updates designed to set a solid foundation for the future and make BTCPay more powerful than ever before.

BTCPay Server 2.0 realigns the internal architecture with the lessons we’ve learned since 2017, positioning us to better support both plugin developers and ourselves as we implement new features. While it may not seem immediately obvious, we expect users to see the full potential of 2.0 in the upcoming months, as the ecosystem continues to thrive and build on top of it.

## 🔑 Key features

- New interface localization
- New sidebar-only navigation
- New onboarding flow for new users and Point Of Sale
- Three new e-commerce integrations (Wix, Odoo & BigCommerce)
- Checkout v2 default (removal of the legacy checkout)
- Branding enhancements

> Full changelog can be found [here](https://github.com/btcpayserver/btcpayserver/releases/tag/v2.0.0).

Reviewer 2: @pavlenex  The github link doesn't exist... getting a 404 page.

Reviewer 3: yet doesn't exist, but when we publish the update it will be.

---
title: "BTCPay Server 2.0: our biggest update yet!"
date: 2024-10-29
author: Pavlenex
category:
  - "releases"
tags:
  - "btcpay-server"
  - "release"
  - "2.0"
coverImage: "/images/btcpay2-0-featured.png"
---

We are thrilled to announce the release of **BTCPay Server 2.0**, our most significant update since the project began in 2017! 

This version is packed with new features, bug fixes, and crucial backend updates designed to set a solid foundation for the future and make BTCPay more powerful than ever before.

BTCPay Server 2.0 realigns the internal architecture with the lessons we’ve learned since 2017, positioning us to better support both plugin developers and ourselves as we implement new features. While it may not seem immediately obvious, we expect users to see the full potential of 2.0 in the upcoming months, as the ecosystem continues to thrive and build on top of it.

## 🔑 Key features

- New interface localization
- New sidebar-only navigation
- New onboarding flow for new users and Point Of Sale
- Three new e-commerce integrations (Wix, Odoo & BigCommerce)

> - Checkout v2 default (depreciation of the legacy checkout)

Reviewer 4: it's not deprecation but removed / discarded?

Author: Fixed https://github.com/btcpayserver/blog/pull/61/commits/ec680a071e3a2030ffa566fcc3bc0aa352b133ea

During the transition from 1.0 to 2.0, larger instances may experience **a few minutes of database migration**. Once this initial migration is complete, BTCPay will be fully operational, though it will continue migrating some data in the background. For instance, some invoices may not immediately appear in reports or searches, but newer data is migrated first, so we expect minimal disruption. 

As an example, our mainnet demo database, containing around 400,000 invoices, took one to two minutes to start and about 15-20 minutes to fully update all invoices in the background.

## 🔄 Updating to 2.0
Updating to version BTCPay Server 2.0 is a one-way process with no option for rolling back. Because of that, we're making it opt-in, requiring that you [SSH into your server](https://docs.btcpayserver.org/FAQ/ServerSettings/#how-to-ssh-into-my-btcpay-running-on-vps) and run two simple commands:

```ssh
git fetch && git checkout 2.0
btcpay-update.sh
```

> Please note that the database migration might take a few minutes, depending on the size of your installation. Enjoy BTCPay 2.0 and let us know your thoughts!

Reviewer: ```suggestion
Please note that the database migration might take a few minutes, depending on the size of your installation. Enjoy BTCPay 2.0 and [let us know your thoughts](https://github.com/btcpayserver/btcpayserver/discussions/6294)!
```

@pavlenex, sorry I have inadvertently removed link with my changes, adding it back.

Reviewer: Since @NicolasDorier approved the approach with 2.0 branch, merging this as well

> Updating to version BTCPay Server 2.0 is a one-way process with no option for rolling back. Because of that, we're making it opt-in, requiring that you [SSH into your server](https://docs.btcpayserver.org/FAQ/ServerSettings/#how-to-ssh-into-my-btcpay-running-on-vps) and run two simple commands:
> 
> ```ssh
> git fetch && git checkout 2.0
> btcpay-update.sh
> ```
> Please note that the database migration might take a few minutes, depending on the size of your installation. Enjoy BTCPay 2.0 and [let us know your thoughts](https://github.com/btcpayserver/btcpayserver/discussions/6294)!

Reviewer: ```suggestion
Updating to BTCPay Server 2.0 is a one-way process with no option for rolling back. Because of that, we're making it opt-in, requiring that you [SSH into your server](https://docs.btcpayserver.org/FAQ/ServerSettings/#how-to-ssh-into-my-btcpay-running-on-vps) and run:

```ssh
git fetch
git checkout 2.0
btcpay-update.sh
```
Please note that the database migration might take a few minutes, depending on the size of your installation. Enjoy BTCPay 2.0 and [let us know your thoughts](https://github.com/btcpayserver/btcpayserver/discussions/6294)!
```

### ✔ Post-update checklist
- If your server **crashed** during the update (it shouldn’t, we’ve done extensive testing), please post server logs [here](https://github.com/btcpayserver/btcpayserver/discussions/6296), we will prioritize support for instances that experienced problems during migration
- If you use **plugins in BTCPay**, not all of them are 2.0 compatible yet. We automatically deactivate the incompatible plugins (communicate this to a plugin dev and kindly ask them to update their plugins, once they’re compatible you can update them)
- If you use **e-commerce integrations** (Woocommerce, Shopify, etc) update those plugins to their latest version in their dedicated stores as they ensure 2.0 compatibility
- If you’re a **developer** leveraging BTCPay’s Greenfield API, check our [breaking changes](https://github.com/btcpayserver/btcpayserver/issues/5964) to ensure 2.0 compatibility 

> - If you’re a **plugin developer**, make sure your plugins are 2.0 compatible. To start, bump Bump plugin to 2.0 submodule and fix errors

Reviewer 4: double bump

Author: Fixed

Author: fixed https://github.com/btcpayserver/blog/pull/61/commits/91d238d6c731eb47198b65eeed26c5139dec1cd3

**All of our official e-commerce integrations are 2.0 compatible.**

However, if you face issues with 2.0 and any official e-commerce integration, please [report it here](https://github.com/btcpayserver/btcpayserver/discussions/6300), and we’ll prioritize those reports. We continue to monitor feedback from the community and ensure existing integrations stay up-to-date. 

If you're using our **Shopify** plugin, rest assured that it’s not impacted by recent API changes, and you can continue to use it without issues. We’re aware of Shopify discounting certain API calls at the end of this year, and we’re already working on shipping a brand new plugin before that happens.

Additionally, our **PrestaShop** plugin received a [new update](https://github.com/btcpayserver/prestashop-plugin/releases/), thanks to our contributor BitcoinMitchell, and the PHP Library has been updated to be [compatible with BTCPay 2.0](https://github.com/btcpayserver/btcpayserver-greenfield-php/releases) as well. 

## 🔌 Plugins

> If you use plugins with your BTCPay Server, you might notice that some are disabled after updates - we do that to prevent server crashes. If you rely heavily on plugins for your BTCPay operations, we recommend holding off on updating to version 2.0 for now. Keep an eye on announcements from developers of the plugins and give them a few more weeks to address compatibility issues and release updates.

Author: @NicolasDorier @Kukks @rockstardev should we still communicate this part?

> If you use plugins with your BTCPay Server, you might notice that some are disabled after updates - we do that to prevent server crashes. If you rely heavily on plugins for your BTCPay operations, we recommend holding off on updating to version 2.0 for now. Keep an eye on announcements from developers of the plugins and give them a few more weeks to address compatibility issues and release updates.

Now most plugins are 2.0 compatible?

Reviewer: ```suggestion
If you use plugins with your BTCPay Server, you may notice that some are disabled following updates - we do this to prevent server crashes. If you rely heavily on plugins for your BTCPay operations, we recommend that you double-check compatibility and look for announcements from plugin developers regarding their support for version 2.0 before proceeding with the upgrade.
```

Currently, maintaining multiple versions of a plugin (for example, one branch for version 1.x and another for version 2.0) is not possible. If you need to publish updates, continue with the 1.x version and release new versions of the plugin until you reach a feature freeze. Once you've achieved this, you can update the submodule reference and release a new version of the plugin starting from 2.0.

We strongly urge all developers to prioritize upgrading to version 2.0.

### List of currently incompatible plugins 

List of currently incompatible plugins, at the time of publishing this blog post, **all other plugins are 2.0 compatible**. 

- BTCPayServer.Plugins.Strike (1.3.2.0)

> - BTCPayServer.Plugins.LiquidPlus (1.1.4.0)

Reviewer 4: Kukks updated Liquid afaics?

Author: Fixed https://github.com/btcpayserver/blog/pull/61/commits/91d238d6c731eb47198b65eeed26c5139dec1cd3
